app/cni/cmd/install/main.go

```diff
@@ -0,0 +1,237 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/natefinch/atomic"
+	"github.com/pkg/errors"
+
+	"github.com/kumahq/kuma/pkg/core"
+	"github.com/kumahq/kuma/pkg/util/files"
+)
+
+const (
+	kumaCniBinaryPath  = "/opt/cni/bin/kuma-cni"
+	primaryBinDir      = "/host/opt/cni/bin"
+	secondaryBinDir    = "/host/secondary-bin-dir"
+	serviceAccountPath = "/var/run/secrets/kubernetes.io/serviceaccount"
+)
+
+var (
+	log = core.NewLoggerWithRotation(2, "/tmp/install-cni.log", 100, 0, 0).WithName("install-cni")
+)
+
+func removeBinFiles() error {
+	return os.Remove("/host/opt/cni/bin/kuma-cni")
+}
+
+func cleanup(ic *InstallerConfig) {
+	log.Info("starting cleanup")
+	if err := removeBinFiles(); err != nil {
+		log.Error(err, "could not remove cni bin file")
+	}
+	log.V(1).Info("removed existing binaries")
+	if err := revertConfig(ic.MountedCniNetDir, ic.CniConfName, ic.ChainedCniPlugin); err != nil {
+		log.Error(err, "could not revert config")
+	}
+	log.V(1).Info("reverted config")
+	if err := removeKubeconfig(ic.MountedCniNetDir, ic.KubeconfigName); err != nil {
+		log.Error(err, "could not remove kubeconfig")
+	}
+	log.V(1).Info("removed kubeconfig")
+	log.Info("finished cleanup")
+}
+
+func removeKubeconfig(mountedCniNetDir, kubeconfigName string) error {
+	kubeconfigPath := mountedCniNetDir + "/" + kubeconfigName
+	if files.FileExists(kubeconfigPath) {
+		err := os.Remove(kubeconfigPath)
+		if err != nil {
+			return errors.Wrap(err, "couldn't remove kubeconfig file")
+		}
+	}
+	return nil
+}
+
+func revertConfig(mountedCniNetDir, cniConfName string, chainedCniPlugin bool) error {
+	configPath := mountedCniNetDir + "/" + cniConfName
+
+	if !files.FileExists(configPath) {
+		log.Info("no need to revert config - file does not exist", "configPath", configPath)
+		return nil
+	}
+
+	if chainedCniPlugin {
+		contents, err := ioutil.ReadFile(configPath)
+		if err != nil {
+			return errors.Wrap(err, "couldn't read cni conf file")
+		}
+		newContents, err := revertConfigContents(contents)
+		if err != nil {
+			return errors.Wrap(err, "could not revert config contents")
+		}
+		err = atomic.WriteFile(configPath, bytes.NewReader(newContents))
+		if err != nil {
+			return errors.Wrap(err, "could not write new conf")
+		}
+	} else {
+		err := os.Remove(configPath)
+		if err != nil {
+			return errors.Wrap(err, "couldn't remove cni conf file")
+		}
+	}
+
+	return nil
+}
+
+func install(ic *InstallerConfig) error {
+	if err := copyBinaries(); err != nil {
+		return err
```

Comment:
here I'd wrap an error. If you are retrowing `err` too much without wrapping, it's then hard to say what is the origin of the error, because we don't have stacktraces.

I like to wrap errors at least when I call "significant big functions" like these in `install` function